lib/orderbook/MatchingEngine.ts

```diff
@@ -209,7 +222,15 @@ class MatchingEngine {
     if (isOwnOrder(order)) {
       return order.isBuy ? this.ownOrders.buy : this.ownOrders.sell;
     } else {
-      return order.isBuy ? this.peerOrders.buy : this.peerOrders.sell;
+      let peerOrdersList = this.peersOrders.get(order.peerPubKey);
+      if (!peerOrdersList) {
+        peerOrdersList = {
+          buy: new Map<orderId, StampedPeerOrder>(),
+          sell: new Map<orderId, StampedPeerOrder>(),
+        };
+        this.peersOrders.set(order.peerPubKey, peerOrdersList);
```

Comment:
I don't think we're trying to add the order to the list in this method. I'm thinking that the list really *should* already exist at this point (having been created when the order was added to it). In fact, the only time we call this method, we call `delete` on the the order immediately after.

I'd probably just have this return `undefined` if `peerOrdersList` is undefined.